sync/class.jetpack-sync-module-full-sync.php

```diff
@@ -87,85 +81,124 @@ function start( $modules = null ) {
 				}
 			}
+
+			if ( $items_sent > 0 ) {
+				$this->update_status_option( "{$module->name()}_sent", $items_sent );
+			}	
 		}
 
 		if ( isset( $actions_with_counts['jetpack_full_sync_end'] ) ) {
-			$status['finished'] = time();
+			$this->update_status_option( "finished", time() );
 		}
-
-		$this->update_status( $status );
 	}
 
 	private function set_status_queuing_started() {
-		$status            = $this->initial_status;
-		$status['started'] = time();
-		$this->update_status( $status );
+		$this->clear_status();
+		$this->update_status_option( "started", time() );
 	}
 
 	private function set_status_queuing_finished() {
-		$this->update_status( array( 'queue_finished' => time() ) );
+		$this->update_status_option( "queue_finished", time() );
 	}
 
-	private $initial_status = array(
-		'started'        => null,
-		'queue_finished' => null,
-		'sent_started'   => null,
-		'finished'       => null,
-		'sent'           => array(),
-		'queue'          => array(),
-	);
+	public function is_started() {
+		return !! $this->get_status_option( "started" );
+	}
 
-	public function get_status() {
-		return get_option( self::STATUS_OPTION, $this->initial_status );
+	public function is_finished() {
+		return !! $this->get_status_option( "finished" );
 	}
 
-	public function update_status( $status ) {
-		return update_option(
-			self::STATUS_OPTION,
-			array_merge( $this->get_status(), $status )
+	public function get_status() {
+		$status = array(
+			'started'        => $this->get_status_option( 'started' ),
+			'queue_finished' => $this->get_status_option( 'queue_finished' ),
+			'sent_started'   => $this->get_status_option( 'sent_started' ),
+			'finished'       => $this->get_status_option( 'finished' ),
+			'sent'           => array(),
+			'queue'          => array(),
 		);
+
+		foreach ( Jetpack_Sync_Modules::get_modules() as $module ) {
+			$queued = $this->get_status_option( "{$module->name()}_queued" );
+			$sent   = $this->get_status_option( "{$module->name()}_sent" );
+
+			if ( $queued > 0 ) {
+				$status[ 'queue' ][ $module->name() ] = $queued;
+			}
+			
+			if ( $sent > 0 ) {
+				$status[ 'sent' ][ $module->name() ] = $sent;
+			}
+		}
+
+		return $status;
 	}
 
 	public function clear_status() {
-		delete_option( self::STATUS_OPTION );
+		$prefix = self::STATUS_OPTION_PREFIX;
+		delete_option( "{$prefix}_started" );
+		delete_option( "{$prefix}_queue_finished" );
+		delete_option( "{$prefix}_sent_started" );
+		delete_option( "{$prefix}_finished" );
+
+		foreach ( Jetpack_Sync_Modules::get_modules() as $module ) {
+			delete_option( "{$prefix}_{$module->name()}_queued" );
+			delete_option( "{$prefix}_{$module->name()}_sent" );
+		}
+	}
+
+	private function get_status_option( $option ) {
+		$prefix = self::STATUS_OPTION_PREFIX;
+		return intval( get_option( "{$prefix}_{$option}", 0 ) );
+	}
+
+	private function update_status_option( $name, $value ) {
+		$prefix = self::STATUS_OPTION_PREFIX;
+		/**
```

Comment:
@gravityrail I'm not sure we really need this, since Jetpack requires WordPress 4.4:
https://github.com/Automattic/jetpack/blob/4.1.1/readme.txt#L5

In general, we support the current version -1.